File: EntityModels.Postgresql/MonsterPart.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EntityModels.Postgresql;

[Table("MonsterPart")]
public partial class MonsterPart
{
    [Key] public string PartName { get; set; } = null!;
    public string PartType { get; set; } = null!;
    public double MaximumHealth { get; set; }
    public double StartingHealth { get; set; }
    public string MonsterName { get; set; } = null!;
}

File: client/src/models/region/MonsterPart.ts
export class MonsterPart {
  constructor(partName: string, partType: string, maximumHealth: number, currentHealth: number) {
    this.partName = partName;
    this.partType = partType;
    this.maximumHealth = maximumHealth;
    this.currentHealth = currentHealth;
  }

  partName: string;
  partType: string;
  maximumHealth: number;
  currentHealth: number;
}